be2-scala/src/test/scala/ch/epfl/pop/pubsub/graph/validator/Lao/CreateLaoContentSuite.scala

```diff
@@ -0,0 +1,56 @@
+package ch.epfl.pop.pubsub.graph.validator.lao
+
+import org.scalatest.{FlatSpec, Matchers}
+import org.scalatest.Outcome
+import util.examples.CreateLaoExamples
+import ch.epfl.pop.pubsub.graph.Validator
+import ch.epfl.pop.model.network.JsonRpcRequest
+import ch.epfl.pop.pubsub.graph.MessageDecoder
+import ch.epfl.pop.pubsub.graph.GraphMessage
+import akka.stream.scaladsl.Flow
+import akka.stream.scaladsl.GraphDSL
+import akka.NotUsed
+import akka.stream.FlowShape
+import ch.epfl.pop.model.network.method.message.Message
+import org.scalatest.GivenWhenThen
+import ch.epfl.pop.model.network.JsonRpcResponse
+import org.scalatest.Inside
+import ch.epfl.pop.model.network.requests.lao.JsonRpcRequestCreateLao
+
+class CreateLaoContentSuite extends FlatSpec with Matchers with Inside with GivenWhenThen {
+
+  /**Decodes data before passing it to the test**/
+  def withCreateLaoFixture(createLaoData: Message)(testCode: GraphMessage => Any) {
+    // Raw encoded data data
+    val message = Left(CreateLaoExamples.getJsonRequestFromMessage(createLaoData))
+    // Decode data
+    val decoded = MessageDecoder.parseData(message)
+    decoded match {
+      case decoded @ Left(_: JsonRpcRequestCreateLao) =>
+        testCode(decoded)
+      case decode @ Left(_) => fail("Decoder decoded to bad type: expeted type is JsonRpcRequestCreateLao")
+      case decoded @ Right(_) =>
+        fail("Message could not be decoded/parsed")
+    }
+  }
+
+  behavior.of("A validator when receiving ")
+  "a CreateLao data with valid content" should "be accepted by validator" in withCreateLaoFixture(CreateLaoExamples.createLao){
+    Given("a valid decoded createLao request ")
+    (msg) => {
+      When("validated")
+      val validated = Validator.validateMessageDataContent(msg)
+      inside(validated){
+        case Left(msg) => {
+          Then("the validation succeeds")
+          msg shouldBe a [JsonRpcRequest]
+        }
+        case _ @ Right(_) => fail("fails to validate a valid CreateLao data content")
+        case _ => fail(s"validated message <$validated> is of unexcpected type")
+      }
+      And("the message has the same content after validation")
+      validated should equal(msg)
+    }
+  }
```

Comment:
This doesn't feel very readable. 

I'm not sure how it can be improved (I'm not a Scala expert), but intuitively I suspect a combination of cleaner formatting / better naming / removing typos / code comments should help.

Reply:
> Nice to see progress on these tests, I look forward to seeing the many that will follow 👍
> 
> However, I would expect somewhat cleaner code at this stage of the semester.
> 
> Code is going to be written once, updated a few times, and read many times. If it's not well formatted, not readable or full of typos, it's going to be tiring to read it and painful to update, which will lead to developer being slower when working on it and/or making mistakes, hence bugs and higher maintenance costs.
> 
> Conversely, clean codes will reduce maintenance costs and make future developers happy, which in turn will make you their hero. Go for it 😉

Sorry to hear that I'll try harder next time.